#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "../protos/Car.proto"
EXTERNPROTO "../protos/VehicleWheel.proto"
EXTERNPROTO "../protos/VehicleLights.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/road/protos/StraightRoadSegment.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/road/protos/RoadIntersection.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/road/protos/CurvedRoadSegment.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/Pavement.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.219857254604728 -0.7463224861136281 -0.6282241115389041 5.406218970916299
  position -25.258273149702646 -19.98863197227554 42.13657517850684
  follow "vehicle"
  followType "None"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF road0 RoadIntersection {
  translation 10 10 0.01
  rotation 0 0 1 0.7854
  name "road0"
  id "0"
  connectedRoadIDs [
    "7"
    "6"
    "5"
    "4"
  ]
  roadsWidth 4.4
  startRoadsLength [
    0
  ]
  boundingObject TRUE
}
DEF road1 StraightRoadSegment {
  translation 22.2 0 0.01
  rotation 0 0 1 1.5707996938995747
  name "road1"
  id "1"
  startJunction "16"
  endJunction "12"
  width 4
  roadBorderHeight 0.5
  startingRoadBorderWidth 0.2
  endingRoadBorderWidth 0.2
  length 7.8
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road2 RoadIntersection {
  translation 10 22.2 0.01
  rotation 0 0 1 0.7854
  name "road2"
  id "2"
  connectedRoadIDs [
    "7"
    "10"
    "9"
  ]
  roadsWidth 4.4
  startRoadsLength [
    0
  ]
  boundingObject TRUE
}
DEF road3 StraightRoadSegment {
  translation 22.2 12.2 0.01
  rotation 0 0 1 1.5707996938995747
  name "road3"
  id "3"
  startJunction "12"
  endJunction "11"
  width 4
  roadBorderHeight 0.5
  startingRoadBorderWidth 0.2
  endingRoadBorderWidth 0.2
  length 7.8
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road4 StraightRoadSegment {
  translation 0 10 0.01
  name "road4"
  id "4"
  endJunction "0"
  width 4
  roadBorderHeight 0.5
  startingRoadBorderWidth 0.2
  endingRoadBorderWidth 0.2
  length 7.8
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road5 StraightRoadSegment {
  translation 10 0 0.01
  rotation 0 0 1 1.5707996938995747
  name "road5"
  id "5"
  endJunction "0"
  width 4
  roadBorderHeight 0.5
  startingRoadBorderWidth 0.2
  endingRoadBorderWidth 0.2
  length 7.8
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road6 StraightRoadSegment {
  translation 12.2 10 0.01
  name "road6"
  id "6"
  startJunction "0"
  endJunction "12"
  width 4
  roadBorderHeight 0.5
  startingRoadBorderWidth 0.2
  endingRoadBorderWidth 0.2
  length 7.8
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road7 StraightRoadSegment {
  translation 10 12.2 0.01
  rotation 0 0 1 1.5707996938995747
  name "road7"
  id "7"
  startJunction "0"
  endJunction "2"
  width 4
  roadBorderHeight 0.5
  startingRoadBorderWidth 0.2
  endingRoadBorderWidth 0.2
  length 7.8
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road8 StraightRoadSegment {
  translation 24.4 10 0.01
  name "road8"
  id "8"
  startJunction "12"
  endJunction "14"
  width 4
  roadBorderHeight 0.5
  startingRoadBorderWidth 0.2
  endingRoadBorderWidth 0.2
  length 7.8
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road9 CurvedRoadSegment {
  translation 16.1 24.4 0.01
  rotation 0 0 1 1.5707996938995747
  name "road9"
  id "9"
  startJunction "2"
  endJunction "13"
  width 4
  roadBorderHeight 0.5
  roadBorderWidth [
    0.2
  ]
  curvatureRadius 6.1
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road10 StraightRoadSegment {
  translation 12.2 22.2 0.01
  name "road10"
  id "10"
  startJunction "2"
  endJunction "11"
  width 4
  roadBorderHeight 0.5
  startingRoadBorderWidth 0.2
  endingRoadBorderWidth 0.2
  length 7.8
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road11 RoadIntersection {
  translation 22.2 22.2 0.01
  rotation 0 0 1 0.7854
  name "road11"
  id "11"
  connectedRoadIDs [
    "13"
    "10"
    "3"
  ]
  roadsWidth 4.4
  startRoadsLength [
    0
  ]
  boundingObject TRUE
}
DEF road12 RoadIntersection {
  translation 22.2 10 0.01
  rotation 0 0 1 0.7854
  name "road12"
  id "12"
  connectedRoadIDs [
    "8"
    "6"
    "3"
    "1"
  ]
  roadsWidth 4.4
  startRoadsLength [
    0
  ]
  boundingObject TRUE
}
DEF road13 CurvedRoadSegment {
  translation 16.1 24.4 0.01
  name "road13"
  id "13"
  startJunction "11"
  endJunction "9"
  width 4
  roadBorderHeight 0.5
  roadBorderWidth [
    0.2
  ]
  curvatureRadius 6.1
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road14 CurvedRoadSegment {
  translation 32.2 1 0.01
  name "road14"
  id "14"
  startJunction "8"
  endJunction "15"
  width 4
  roadBorderHeight 0.5
  roadBorderWidth [
    0.2
  ]
  curvatureRadius 9
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road15 CurvedRoadSegment {
  translation 31.2 1 0.01
  rotation 0 0 1 -1.5707996938995747
  name "road15"
  id "15"
  startJunction "14"
  endJunction "16"
  width 4
  roadBorderHeight 0.5
  roadBorderWidth [
    0.2
  ]
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF road16 CurvedRoadSegment {
  translation 31.2 0 0.01
  rotation 0 0 1 -3.141592653589793
  name "road16"
  id "16"
  startJunction "15"
  endJunction "1"
  width 4
  roadBorderHeight 0.5
  roadBorderWidth [
    0.2
  ]
  curvatureRadius 9
  roadBoundingObject TRUE
  rightBorderBoundingObject TRUE
  leftBorderBoundingObject TRUE
}
DEF wall1 Transform {
  translation 10 -0.1 0.26
  children [
    Shape {
      appearance Pavement {
      }
      geometry Box {
        size 4.4 0.2 0.5
      }
      castShadows FALSE
    }
  ]
}
DEF wall2 Transform {
  translation -0.1 10 0.26
  rotation 0 0 1 1.5707996938995747
  children [
    Shape {
      appearance Pavement {
      }
      geometry Box {
        size 4.4 0.2 0.5
      }
      castShadows FALSE
    }
  ]
}
DEF wall3 Transform {
  translation 7.9 22.2 0.26
  rotation 0 0 1 1.5707996938995747
  children [
    Shape {
      appearance Pavement {
      }
      geometry Box {
        size 4.4 0.2 0.5
      }
      castShadows FALSE
    }
  ]
}
DEF wall4 Transform {
  translation 24.3 22.2 0.26
  rotation 0 0 1 1.5708003061004252
  children [
    Shape {
      appearance Pavement {
      }
      geometry Box {
        size 4.4 0.2 0.5
      }
      castShadows FALSE
    }
  ]
}
Robot {
  name "super"
  controller "my_supervisor"
  supervisor TRUE
}
Car {
  translation 15.75 9.25 0.208995
  rotation 0 0 1 2.85387e-06
  controller "my_controller"
  supervisor TRUE
  extensionSlot [
    DistanceSensor {
      translation -0.1 0.2 0
      rotation 0 0 1 2.8798
      children [
        Shape {
          geometry Box {
            size 0.03 0.03 0.03
          }
        }
      ]
      name "dsRearLeft"
      lookupTable [
        0.2 1 0
        0.8 0 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation -0.1 -0.2 0
      rotation 0 0 -1 2.87979
      children [
        Shape {
          geometry Box {
            size 0.03 0.03 0.03
          }
        }
      ]
      name "dsRearRight"
      lookupTable [
        0.2 1 0
        0.8 0 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0.5 0.2 0
      rotation 0 0 -1 -1.5708053071795867
      children [
        Shape {
          geometry Box {
            size 0.03 0.03 0.03
          }
        }
      ]
      name "dsLeft"
      lookupTable [
        0.2 1 0
        0.8 0 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 0.5 -0.2 0
      rotation 0 0 -1 1.5707996938995747
      children [
        Shape {
          geometry Box {
            size 0.03 0.03 0.03
          }
        }
      ]
      name "dsRight"
      lookupTable [
        0.2 1 0
        0.8 0 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 1.15 0.2 0
      rotation 0 0 -1 -0.5236053071795865
      children [
        Shape {
          geometry Box {
            size 0.03 0.03 0.03
          }
        }
      ]
      name "dsFrontLeft"
      lookupTable [
        0.2 1 0
        0.8 0 0
      ]
      type "laser"
    }
    DistanceSensor {
      translation 1.15 -0.2 0
      rotation 0 0 -1 0.5235996938995747
      children [
        Shape {
          geometry Box {
            size 0.03 0.03 0.03
          }
        }
      ]
      name "dsFrontRight"
      lookupTable [
        0.2 1 0
        0.8 0 0
      ]
      type "laser"
    }
    Compass {
      translation 0.5 0 0
    }
    GPS {
      translation 0.5 0 0
    }
    DEF LeftSpoiler Transform {
      translation -0.13 0.15 0.2
      rotation 0 1 0 1.0472
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.75 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.04 0.04
          }
        }
      ]
    }
    DEF RightSpoiler Transform {
      translation -0.13 -0.15 0.2
      rotation 0 1 0 1.0472
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.75 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.04 0.04
          }
        }
      ]
    }
    DEF Spoiler Transform {
      translation -0.17 0 0.25
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.75 1 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.15 0.4 0.04
          }
        }
      ]
    }
    DEF RearLeftCorner Transform {
      translation 0.2 0.125 0.225
      scale 3.9 0.77 0.7
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            roughness 1
            metalness 0
          }
          geometry Cone {
            bottomRadius 0.07
            height 0.24
            subdivision 32
          }
        }
      ]
    }
    DEF RearRightCorner Transform {
      translation 0.2 -0.125 0.225
      scale 3.9 0.77 0.7
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            roughness 1
            metalness 0
          }
          geometry Cone {
            bottomRadius 0.07
            height 0.24
            subdivision 32
          }
        }
      ]
    }
    DEF FrontLeftCorner Transform {
      translation 0.7 0.125 0.225
      scale 6.2 0.68 0.7
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            roughness 1
            metalness 0
          }
          geometry Cone {
            bottomRadius 0.08
            height 0.25
            subdivision 32
          }
        }
      ]
    }
    DEF FrontRightCorner Transform {
      translation 0.7 -0.125 0.225
      scale 6.2 0.68 0.7
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            roughness 1
            metalness 0
          }
          geometry Cone {
            bottomRadius 0.08
            height 0.25
            subdivision 32
          }
        }
      ]
    }
    DEF LeftGlass Transform {
      translation 0.45 0.105 0.21
      rotation 1 0 0 0.3
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.495 0.1 0.18
          }
        }
      ]
    }
    DEF RightGlass Transform {
      translation 0.45 -0.105 0.21
      rotation 1 0 0 -0.3
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.49 0.1 0.18
          }
        }
      ]
    }
    DEF RearGlass Transform {
      translation 0.11 0 0.195
      rotation 0 1 0 1.0191003061004251
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.1 0.249 0.28
          }
        }
      ]
    }
    DEF FrontGlass Transform {
      translation 0.8 0 0.17
      rotation 0 1 0 -1.24
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.2 0.245 0.29
          }
        }
      ]
    }
    DEF FrontBumper Transform {
      translation 1 0 0.1
      rotation 1 0 0 -1.5707953071795862
      scale 4 2 1
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.41
            radius 0.06
          }
        }
      ]
    }
    DEF RearBumper Transform {
      translation -0.07 0 0.1
      rotation 1 0 0 -1.5707953071795862
      scale 1.3 1.5 1
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            roughness 1
            metalness 0
          }
          geometry Cylinder {
            height 0.41
            radius 0.07
          }
        }
      ]
    }
    DEF MainBody Transform {
      translation 0.5 0 0.09
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1.2 0.4 0.15
          }
        }
      ]
    }
    DEF TopBody Transform {
      translation 0.45 0 0.263
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.666667 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.5 0.25 0.1
          }
        }
      ]
    }
    VehicleLights {
    }
  ]
  boundingObject Transform {
    translation 0.5 0 0.15
    children [
      Box {
        size 1.5 0.41 0.35
      }
    ]
  }
  physics Physics {
    density -1
    mass 1
  }
  wheelFrontRight VehicleWheel {
    name "front right wheel"
    physics Physics {
      density -1
      mass 1
    }
  }
  wheelFrontLeft VehicleWheel {
    name "front left wheel"
    physics Physics {
      density -1
      mass 1
    }
  }
  wheelRearRight VehicleWheel {
    name "rear right wheel"
    physics Physics {
      density -1
      mass 1
    }
  }
  wheelRearLeft VehicleWheel {
    name "rear left wheel"
    physics Physics {
      density -1
      mass 1
    }
  }
}
